pylint/checkers/classes/class_checker.py

```diff
@@ -874,6 +884,55 @@ def _check_consistent_mro(self, node: nodes.ClassDef) -> None:
         except astroid.DuplicateBasesError:
             self.add_message("duplicate-bases", args=node.name, node=node)
 
+    def _check_enum_base(self, node: nodes.ClassDef, ancestor: nodes.ClassDef) -> None:
+        members = ancestor.getattr("__members__")
+        if members and isinstance(members[0], nodes.Dict) and members[0].items:
+            self.add_message(
+                "invalid-enum-extension",
+                args=ancestor.name,
+                node=node,
+                confidence=INFERENCE,
+            )
+
+        if ancestor.is_subtype_of("enum.IntFlag"):
+            # Collect integer flag assignments present on the class
+            assignments = {
+                assign_name.parent.value.value: assign_name
+                for assign_name in node.nodes_of_class(nodes.AssignName)
+                if isinstance(assign_name.parent, nodes.Assign)
+                and isinstance(getattr(assign_name.parent.value, "value", None), int)
+            }
+
+            # For each bit position, collect all the flags that set the bit
+            bit_flags = defaultdict(set)
+            for flag_value in assignments.keys():
+                for position, bit in enumerate(reversed(f"{flag_value:b}")):
+                    if bit == "1":
+                        bit_flags[position].add(flag_value)
+
+            # When multiple flags overlap on a bit position, pair the min-and-max
+            overlaps = (
+                (min(*flag_values), max(*flag_values))
```

Comment:
In hindsight, this is slightly odd: why only report the maximum overlapping value when multiple items share a common bit position?

(todo: more test coverage to examine this)

Reply:
There was indeed a problem here, and it's now resolved (with test coverage).

Basically, if `1, 5, 9` (for example) were declared as flag values, then the min/max logic that existed here previously would have resulting in the `9` as having a reported overlap, but would not have reported about the `5`.

In practice anyone who noticed that by using `pylint` would probably have run the check again after fixing the `9`, and _then_ been told about the `5`, but that'd be kinda an annoying workflow.

I think the pull request is ready for review again.